### PR TITLE
[CHI-214] Add javascript for modal closing via X icon and escape key

### PR DESCRIPTION
**Pull request by davidojedacl**
@mattnickles @jllr @dani-cl-madrid

### CODE REVIEW

**Comment by mattnickles** [comment]
@davidojedacl can you resolve

File: src/chi/javascript/components/modal.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      this.show();
+    }
+  }
+
+  dispose() {
+    this._config = null;
+    this._modalElem = null;
+    this._shown = null;
+    this._transitioning = null;
+    this._elem.removeEventListener('click', this._triggerClickEventListener);
```
Comment (dani-cl-madrid) [inline src/chi/javascript/components/modal.js]: Keyup event listener should be removed from document.body. You should use method this._addEventHandler from the Component class (Modal inherits this class) to add all event listeners and then you'll only need to call this._removeEventHandlers() to remove all at once. This ensures you don't forget to remove any event listener.

File: src/chi/components/modal/modal.scss [hunk 2]
```diff
@@ -15,29 +17,92 @@
...
+      box-shadow: inset 0 -1px 0 0 rgba(set-color(black), 0.12);
+      min-height: 3.5rem;
+      padding: 1rem;
+      position: relative;
+
+      &.-noBorder {
+        box-shadow: none;
+      }
+
+      .a-modal__title {
```
Comment (mattnickles) [inline src/chi/components/modal/modal.scss]: @davidojedacl doesn't look like this is getting applied

Comment (davidojedacl) [reply src/chi/components/modal/modal.scss]: i will check

**Comment by davidojedacl** [comment]
OK. i set global z-index an d poiotion relative related examples and i updated the z-index strategy

**Pull request merged**